x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/voyageai/VoyageAIModel.java

```diff
@@ -73,22 +88,20 @@ public SecureString apiKey() {
         return apiKey;
     }
 
-    public VoyageAIRateLimitServiceSettings rateLimitServiceSettings() {
-        return rateLimitServiceSettings;
-    }
+    public int rateLimitGroupingHash() {
+        String modelId = getServiceSettings().modelId();
+        String modelFamily = MODEL_TO_MODEL_FAMILY.getOrDefault(modelId, DEFAULT_MODEL_FAMILY);
 
-    public abstract ExecutableAction accept(VoyageAIActionVisitor creator, Map<String, Object> taskSettings, InputType inputType);
+        return Objects.hash(modelFamily, apiKey);
```

Comment:
This is a notable change. Previously the request manager was not including the `apiKey` in the grouping. This didn't seem right to me though because it'd mean that all users who were using the same model id family would be rate limited together. From the voyageai docs it does seem like you can have more granular rate limits per project. This doesn't accomplish that but at least it's a step in that direction because we shouldn't be grouping all users together.